geps/gep-1619.md

```diff
@@ -163,43 +163,258 @@ Java application servers such as Tomcat and Jetty, were the first to standardize
+- Name=_value_
 - Expires=_date_
 - Max-Age=_number_
 - Domain=_domain_
 - Path=_path-value_
 - Secure
 - HttpOnly
 
-However, there are de-facto standard attributes that have been proposed as updates to [rfc6265](https://www.rfc-editor.org/rfc/rfc6265#section-5.2):
-- SameSite=_[Strict|Lax|None]_ ([Draft RFC Update](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-cookie-same-site-00))
-- Partitioned ([Draft RFC Update](https://www.ietf.org/archive/id/draft-cutler-httpbis-partitioned-cookies-01.html))
-
-**Session Cookie Strategy**
-
-Session cookie strategy is configuration specific to the creation of the cookie by the dataplane. These are not specific configuration parameters of the cookie itself, but instead the operation to take when inserting the cookie such as:
-- Rewriting an existing cookie value
-- Inserting a new cookie
-- Prefixing an existing cookie
-- Preserving an existing cookie value
-- Only adding cookies on specific HTTP methods (POST/GET)
+Other cookie attributes not defined by RFC6265, but are captured in draft RFCs and could be considered de facto
+standards due to wide acceptance are:
+- SameSite=[Strict|Lax|None]
+- Partitioned
+
+We implement core support for the `Name`, `Path`, and `TTL` attributes as the metrics in [implementations](#Implementations)
+for session persistence shows these being the most widely adopted and supported. Subsequently, we proceed with `Domain`,
+`SameSite`, `HttpOnly`, and `Secure` as extended support, considering their partial acceptance among APIs. The
+`Partitioned` attribute lacks support in any implementation API and will be omitted. Similarly, since the `Expires`
+attribute is redundant compared to the preferred `Max-Age` attribute, it will also be skipped.
+
+```go
+// SessionPersistencePolicy provides a way to define session persistence rules
+// for a service.
+//
+// Support: Core
+type SessionPersistencePolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+    
+    // Spec defines the desired state of SessionPersistencePolicy.
+    Spec SessionPersistencePolicySpec `json:"spec"`
+    
+    // Status defines the current state of SessionPersistencePolicy.
+    Status SessionPersistencePolicyStatus `json:"status,omitempty"`
+}
+
+// SessionPersistencePolicySpec defines the desired state of
+// SessionPersistencePolicy.
+// Note: there is no Override or Default policy configuration.
+type SessionPersistencePolicySpec struct {
+    // TargetRef identifies an API object to apply policy to.
+    // Services are the only valid API target references.
+    TargetRef gatewayv1a2.PolicyTargetReference `json:"targetRef"`
+    
+    // HttpCookie defines and configures a cookie to achieve
+    // session persistence. 
+    HttpCookie *HttpCookie `json:"httpCookie"`
+}
+
+// HttpCookie defines a cookie to achieve session persistence.
+//
+// Support: Core
+type HttpCookie struct {
+    // Name defines the cookie's name.
```

Comment:
Can we make this optional and allow an implementation to specify a default value?